circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/annotation/TableAnnotatorFactory.java

```diff
@@ -0,0 +1,23 @@
+package com.hotels.bdp.circustrain.core.annotation;
+
+import com.hotels.bdp.circustrain.api.conf.OrphanedDataStrategy;
+import com.hotels.bdp.circustrain.api.conf.ReplicationMode;
+
+public class TableAnnotatorFactory {
+
+  public static TableAnnotator newInstance(ReplicationMode replicationMode,
+    OrphanedDataStrategy orphanedDataStrategy) {
+    if (replicationMode == ReplicationMode.FULL) {
+      if (orphanedDataStrategy == OrphanedDataStrategy.BEEKEEPER) {
```

Comment:
I think if you want to use BEEKEEPER you need to set the OrphanedDataStrategy to NULL/NONE for CT and just add the table properties in a "generic" way.

Reply:
In this case we probably don't need `OrphanedDataStrategy` at all, just a housekeeping "on" or "off" option. Then `OrphanedDataOptions` is really just `ReplicaTableAnnotations` or something, and the user can just add whatever they like, including Beekeeper parameters which they can find in the docs.

Reply:
I've updated this to three options for `OrphanedDataStrategy`: `HIVE_HOOK`, `HOUSEKEEPING` and `NONE` where `HIVE_HOOK` should add the options specified in `OrphanedDataOptions`.

What do people think about the name `HIVE_HOOK`? Trying to find a name that summarises the cleanup mechanism using hive events, but happy to change it.

Reply:
Should not have it imo :). OrphanedDataOptions should be a more generic add table properties option. CT is just setting a table property it shouldn't have to know it sets a beekeeper property that's up to the user. HIVE_HOOK is the same as Beekeeper. From a user perspective if I want to use CT with beekeeper. I need to add the table property and turn off housekeeping. CT doesn't know about beekeeper nor about hive events.

Reply:
The thing is don't we only want to add table annotations (e.g. for Beekeeper) if it's in `FULL` replication mode - i.e. not just metadata update? In which case CT would need to know to only add table params when its in a specific housekeeping mode.

Or do we presume, like with Beekeeper, that metadata only updates are handled properly? In which case, yep we can just have `NONE` and `HOUSEKEEPING` and change `OrphanedDataOptions` to `ReplicaTableAnnotations` or something.

Reply:
If that's the case, the user should remove the table parameters from the configuration. Beekeeper and CT are completely separate we can't really have any logic in CT that says oh if we are beekeeping let's do this or that. That would tie beekeeper and CT together and we won't be able to change one without affecting the other. The only thing driving beekeeping is having a property on the table, you can add that by hand or we provide a mechanism to add table properties in CT which is a bit nicer.

Reply:
As in, if the update is just a metadata update, at the moment we currently don't schedule a path for deletion in housekeeping. I thought that this would mean we also wouldn't want to emit any event with a beekeeper param.

But beekeeper handles metadata updates (by ignoring them), so maybe we dont care about this. We just want all events to be emitted with a table param that the user sets, and then whatever downstream cleaner picks it up can decide whether or not to do cleanup.

I think I follow you now, good plan. Will remove references to `HIVE_HOOK` and just have two strategy options. This makes things simpler :)

Reply:
I guess if it is a metadata only update beekeeper will no location has  been changed so nothing to delete right.